File: app/src/main/java/com/wolcano/musicplayer/music/mvp/listener/OnSwipeTouchListener.java
package com.wolcano.musicplayer.music.mvp.listener;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import com.wolcano.musicplayer.music.provider.RemotePlay;


public class OnSwipeTouchListener implements View.OnTouchListener {

    private final GestureDetector gestureDetector;
    private Context context;
    protected OnSwipeTouchListener(Context context) {
        this.context = context;
        gestureDetector = new GestureDetector(context, new GestureListener(this));
    }
    public void onSwipeLeft() {
        RemotePlay.get().next(context,false);
    }

    public void onSwipeRight() {
        RemotePlay.get().prev(context);
    }
    public void onClick() {

    }
    public boolean onTouch(View v, MotionEvent event) {
        return gestureDetector.onTouchEvent(event);
    }

    private final class GestureListener extends GestureDetector.SimpleOnGestureListener {

        private final int SWIPE_DISTANCE_THRESHOLD = 100;
        private final int SWIPE_VELOCITY_THRESHOLD = 100;
        private OnSwipeTouchListener mHelper;

        GestureListener(OnSwipeTouchListener helper) {
            mHelper = helper;
        }
        @Override
        public boolean onSingleTapUp(MotionEvent e) {
            mHelper.onClick();
            return true;
        }
        @Override
        public boolean onDown(MotionEvent e) {
            return true;
        }


        @Override
        public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
            float distanceX = e2.getX() - e1.getX();
            float distanceY = e2.getY() - e1.getY();
            if (Math.abs(distanceX) > Math.abs(distanceY) && Math.abs(distanceX) > SWIPE_DISTANCE_THRESHOLD && Math.abs(velocityX) > SWIPE_VELOCITY_THRESHOLD) {
                if (distanceX > 0)
                    onSwipeRight();
                else
                    onSwipeLeft();
                return true;
            }
            return false;
        }
    }
}

File: app/src/main/java/com/wolcano/musicplayer/music/listener/OnSwipeTouchListener.kt
package com.wolcano.musicplayer.music.listener

import android.content.Context
import android.view.GestureDetector
import android.view.GestureDetector.SimpleOnGestureListener
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import com.wolcano.musicplayer.music.provider.RemotePlay.next
import com.wolcano.musicplayer.music.provider.RemotePlay.prev
import kotlin.math.abs

open class OnSwipeTouchListener(private val context: Context) : OnTouchListener {

    private var gestureDetector: GestureDetector? = null

    init {
        gestureDetector = GestureDetector(context, GestureListener(this))
    }

    open fun onSwipeLeft() {
        next(context, false)
    }

    open fun onSwipeRight() {
        prev(context)
    }

    open fun onClick() {}

    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
        return gestureDetector!!.onTouchEvent(event)
    }

    private inner class GestureListener(private val mHelper: OnSwipeTouchListener) :
        SimpleOnGestureListener() {
        private val SWIPE_DISTANCE_THRESHOLD = 100
        private val SWIPE_VELOCITY_THRESHOLD = 100
        override fun onSingleTapUp(e: MotionEvent): Boolean {
            mHelper.onClick()
            return true
        }

        override fun onDown(e: MotionEvent): Boolean {
            return true
        }

        override fun onFling(
            e1: MotionEvent,
            e2: MotionEvent,
            velocityX: Float,
            velocityY: Float
        ): Boolean {
            val distanceX = e2.x - e1.x
            val distanceY = e2.y - e1.y
            if (abs(distanceX) > abs(distanceY) && abs(distanceX) > SWIPE_DISTANCE_THRESHOLD && abs(
                    velocityX
                ) > SWIPE_VELOCITY_THRESHOLD
            ) {
                if (distanceX > 0) onSwipeRight() else onSwipeLeft()
                return true
            }
            return false
        }
    }

}